packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -568,14 +670,22 @@ export class SeedlessOnboardingController extends BaseController<
       // Note that vault encryption using the password is a very costly operation as it involves deriving the encryption key
       // from the password using an intentionally slow key derivation function.
       // We should make sure that we only call it very intentionally.
-      updatedState.vault = await this.#vaultEncryptor.encrypt(
-        password,
-        serializedVaultData,
-      );
+      const { vault, exportedKeyString } =
+        await this.#vaultEncryptor.encryptWithDetail(
+          password,
+          serializedVaultData,
+        );
+      updatedState.vault = vault;
+      updatedState.vaultEncryptionKey = exportedKeyString;
+      updatedState.vaultEncryptionSalt = JSON.parse(vault).salt;
 
       this.update((state) => {
         state.vault = updatedState.vault;
+        state.vaultEncryptionKey = updatedState.vaultEncryptionKey;
+        state.vaultEncryptionSalt = updatedState.vaultEncryptionSalt;
       });
+
+      this.#password = password;
```

Comment:
since we are setting `vaultEncryptionKey` above, do we really need to set `password` as well in class variable, wht the case when `vaultEncryptionKey` might be missing but password would be required instead?